python/tests/test_random_forest.py

```diff
@@ -354,6 +356,27 @@ def test_random_forest_classifier(
         else:
             assert cu_acc - spark_acc < 0.07
 
+        # for multi-class classification evaluation
+        if n_classes > 2:
+            from pyspark.ml.evaluation import MulticlassClassificationEvaluator
+
+            evaluator = MulticlassClassificationEvaluator(
+                predictionCol=spark_rf_model.getPredictionCol(),
+                labelCol=spark_rf_model.getLabelCol(),
+            )
+
+            spark_cuml_f1_score = spark_rf_model.transformEvaluate(test_df)
+            transfomred_df = spark_rf_model.transform(test_df)
+            prediction_col_name = spark_rf_model.getPredictionCol()
+
+            # MulticlassClassificationEvaluator requires prediction col with double type
+            transformed_df = transfomred_df.withColumn(
+                prediction_col_name, transfomred_df[prediction_col_name].cast("double")
+            )
+            pyspark_f1_score = evaluator.evaluate(transformed_df)
+
+            assert math.fabs(pyspark_f1_score - spark_cuml_f1_score) < 1e-6
```

Comment:
transformEvaluate seems equivalent to "transform + evaluate (line 369-line 376)". Is it better to move transformEvaluate from model class to perhaps util.py? transformEvaluate seems not in the method list of Pyspark model class, according to https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.ml.classification.RandomForestClassificationModel.html#pyspark.ml.classification.RandomForestClassificationModel

Reply:
Hmm, good point. But transformEvaluate is a customized API to be used internally and it should be different implementation for different subclass.